sphinx/themes/basic/static/searchtools.js

```diff
@@ -547,7 +547,7 @@ const Search = {
 
         // set score for the word in each file
         recordFiles.forEach((file) => {
-          if (!scoreMap.has(file)) scoreMap.set(file, {});
+          if (!scoreMap.has(file)) scoreMap.set(file, new Map());
           scoreMap.get(file)[word] = record.score;
```

Comment:
Per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#setting_object_properties

```suggestion
          if (!scoreMap.has(file)) scoreMap.set(file, new Map());
          const file_scores = scoreMap.get(file);
          file_scores.set(word, record.score);
```

Reply:
Also Line 590 needs updating:

```diff
-      const score = Math.max(...wordList.map((w) => scoreMap.get(file)[w]));
+      const score = Math.max(...wordList.map((w) => scoreMap.get(file).get(w)));

Reply:
We may need to be careful about updating the corresponding score retrieval alongside this: https://github.com/sphinx-doc/sphinx/blob/2a1a43ce4b4729795e8bd5bdcaa34488bb407d83/sphinx/themes/basic/static/searchtools.js#L590

(also, a nitpick preference for `fileScores` instead of `file_scores`; but that's less important)

Reply:
Indeed, and use whichever name you prefer.